### ISSUE TITLE
RegenFarms - let's make it so that the "harvest rewards button" triggers the same tx.

### ISSUE THREAD

**Issue by laurenluz** [Giveth/giveth-dapps-v2#817]
Right now, there are two "harvest" buttons: one from GIVfarm triggers the tx to claim all your rewards... one triggers a tx to claim just the rewards liquid from the stream.

This doesn't really make sense on mainnet... we should make the trigger a harvest all from either button to save the user needless gas.

@mateodaza @aminlatifi  does this make sense? I would love to fix this for both the FOX & CULT farms

![image.png](https://images.zenhubusercontent.com/5fdfac16a58523a866ddb68d/66617de5-5d1c-4956-9cae-5fcbca321d5f)

**Comment by aminlatifi** [comment]
@laurenluz You are right, we can save the user gas by triggering farm harvest instead of the stream's. But there are few points:
1. What happens if a regan farm has multiple farms! In this case, we cannot do it anymore since we don't know trigger which one!
2. The user may intentionally like to harvest the stream alone since it needs less gas and there isn't much to harvest from the farm.
**Gas used by harvesting stream**
<img width="478" alt="image" src="https://user-images.githubusercontent.com/5684607/170861910-258b14fd-ea23-4098-a712-7c7fd9cde0df.png">

**Gas used by harvesting a farm**
<img width="503" alt="image" src="https://user-images.githubusercontent.com/5684607/170861950-bdff2e3e-46fb-4312-8c3a-35d940970176.png">


3. Having this feature adds a little to system complexity

**Comment by laurenluz** [comment]
Maybe we should add something to the frontend, a tooltip or warning modal that explains that?

**Comment by aminlatifi** [comment]
> Maybe we should add something to the frontend, a tooltip or warning modal that explains that?
Yeah, it could be a solution. Do you have any offer for the text?

**Comment by laurenluz** [comment]
Yeah, maybe the text shows up as a tooltip over the "Harvest" button on the stream side *if* the use has liquid rewards in both places.

![image](https://user-images.githubusercontent.com/75490971/171010370-24bbcc7b-0e34-49c9-9d19-163b4606a8c6.png)

It could say:
"Use the Harvest button on the farming rewards card to harvest all your liquid CULT in a single transaction."

What do you think @aminlatifi ?

**Comment by aminlatifi** [comment]
> Yeah, maybe the text shows up as a tooltip over the "Harvest" button on the stream side _if_ the use has liquid rewards in both places.
> 
> ![image](https://user-images.githubusercontent.com/75490971/171010370-24bbcc7b-0e34-49c9-9d19-163b4606a8c6.png)
> 
> It could say: "Use the Harvest button on the farming rewards card to harvest all your liquid CULT in a single transaction."
> 
> What do you think @aminlatifi ?

That's good. I can amend it and make this:
"Use the Harvest button on the farming rewards card to harvest both your farming reward and liquid CULT in a single transaction."

**Comment by aminlatifi** [comment]
What do you think @laurenluz

**Comment by aminlatifi** [comment]
@laurenluz No, I see it today
<img width="1004" alt="image" src="https://user-images.githubusercontent.com/5684607/171364063-590ff939-3657-4df0-8d16-3c2264283470.png">

**Comment by laurenluz** [comment]
I don't understand @aminlatifi - there is still not tooltip?

**Comment by MoeNick** [comment]
@laurenluz would u pls coordinate this one with @mosaeedi
